src/vos/evtree.c

```diff
@@ -1930,13 +1901,25 @@ evt_ent_array_fill(struct evt_context *tcx, enum evt_find_opc find_opc,
 			}
 			V_TRACE(DB_TRACE, "Found overlapped leaf rect\n");
 
+			desc = evt_node_desc_at(tcx, node, i);
+			rc = evt_dtx_check_availability(tcx, desc->dc_dtx,
+							intent);
+			/* Skip the unavailable record. */
+			if (rc == ALB_UNAVAILABLE)
```

Comment:
should we return dirty data?

Reply:
That depends on the 'intent' parameter. If it is for aggregation, the garbage dirty will be returned also, otherwise, the evt_dtx_check_availability() will return ALB_UNAVAILABLE for dirty case.

Reply:
> so we have more than 4K in stack in one function, which is not very good, we have to pay more attention IMO.

The current master already exceeds 4K stack.

Reply:
> I think both RC_INPROGRESS and RC_UNAVAIL can be just an error code and only internally checked right? we probably should just use error code.

PROBE_RC_UNAVAILABLE is NOT an error code, it means the current data record cannot be used for invisible to current operation (intent)